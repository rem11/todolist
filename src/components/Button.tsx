import React from 'react';
import {
  StyleProp,
  StyleSheet,
  Text,
  TouchableOpacity,
  ViewStyle,
} from 'react-native';
import {commonStyles} from '../const/styles';
import {useTheme} from '@react-navigation/native';

export type Props = {
  style?: StyleProp<ViewStyle> | undefined;
  onPress: () => void;
  title: string;
};

function Button({style, onPress, title}: Props): JSX.Element {
  const {colors} = useTheme();
  return (
    <TouchableOpacity
      style={[
        styles.button,
        {
          backgroundColor: colors.primary,
        },
        style,
      ]}
      onPress={onPress}>
      <Text style={[commonStyles.text, styles.text]}>{title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  button: {
    padding: 8,
    borderRadius: 12,
  },
  text: {
    color: '#fff',
  },
});

export default Button;
