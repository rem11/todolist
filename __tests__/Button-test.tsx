import 'react-native';
import React from 'react';
import Button from '../src/components/Button';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('button renders correctly', () => {
  expect(
    renderer.create(<Button title="Test button" onPress={() => {}} />),
  ).toMatchSnapshot();
});

test('button renders correctly with style attr', () => {
  expect(
    renderer.create(
      <Button title="Test button" onPress={() => {}} style={{marginTop: 16}} />,
    ),
  ).toMatchSnapshot();
});
