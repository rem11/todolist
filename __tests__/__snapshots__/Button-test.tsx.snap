// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`button renders correctly 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "rgb(0, 122, 255)",
      "borderRadius": 12,
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#fff",
        },
      ]
    }
  >
    Test button
  </Text>
</View>
`;

exports[`button renders correctly with style attr 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "rgb(0, 122, 255)",
      "borderRadius": 12,
      "marginTop": 16,
      "opacity": 1,
      "padding": 8,
    }
  }
>
  <Text
    style={
      [
        {
          "fontSize": 16,
          "lineHeight": 24,
        },
        {
          "color": "#fff",
        },
      ]
    }
  >
    Test button
  </Text>
</View>
`;
